src/NzbDrone.Common/Http/HttpUri.cs

```diff
@@ -144,23 +144,38 @@ private static string CombinePath(string basePath, string relativePath)
                 return basePath;
             }
 
-            var newPath = "/" + relativePath.TrimStart('/');
+            if (basePath.IsNullOrWhiteSpace())
+            {
+                return relativePath;
+            }
+
+            return basePath.TrimEnd('/') + "/" + relativePath.TrimStart('/');
+        }
 
-            if (basePath != null && !relativePath.StartsWith("/"))
+        private static string CombineRelativePath(string basePath, string relativePath)
+        {
+            if (relativePath.IsNullOrWhiteSpace())
             {
-                var baseSlashIndex = basePath.LastIndexOf('/');
+                return basePath;
+            }
 
-                if (baseSlashIndex == basePath.Length - 1)
-                {
-                    newPath = basePath.TrimEnd('/') + newPath;
-                }
-                else if (baseSlashIndex != 0)
-                {
-                    newPath = basePath.Substring(0, baseSlashIndex) + newPath;
-                }
+            if (relativePath.StartsWith("/"))
+            {
+                return relativePath;
             }
 
-            return newPath;
+            var baseSlashIndex = basePath.LastIndexOf('/');
+
+            if (baseSlashIndex == 0)
```

Comment:
I'm confused by this, unless we have a trailing slash or multiple slashes we're always going to throw out the base path?